src/main/java/com/labsynch/labseer/chemclasses/rdkit/CmpdRegMoleculeRDKitImpl.java

```diff
@@ -0,0 +1,195 @@
+import org.springframework.beans.factory.annotation.Configurable;
+
+@Configurable
+public class CmpdRegMoleculeRDKitImpl implements CmpdRegMolecule {
+
+	Logger logger = LoggerFactory.getLogger(CmpdRegMoleculeRDKitImpl.class);
+
+	@Autowired
+	private PropertiesUtilService propertiesUtilService;
+
+	RWMol molecule;
+
+	static {
+		System.loadLibrary("GraphMolWrap");
+	}
+
+	public CmpdRegMoleculeRDKitImpl(String molStructure) throws CmpdRegMolFormatException {
+		this.molecule = RWMol.MolFromMolBlock(molStructure);
+	}
+
+	public CmpdRegMoleculeRDKitImpl(RWMol molecule) throws CmpdRegMolFormatException {
+		this.molecule = molecule;
+	}
+
+	public CmpdRegMoleculeRDKitImpl(ROMol readOnlyMol) {
+		if (readOnlyMol == null) {
+			logger.info("Got empty mol record");
+		} else {
+			this.molecule = RWMol.MolFromMolBlock(readOnlyMol.MolToMolBlock());
+		}
+	}
+
+	@Override
+	public void setProperty(String key, String value) {
+		molecule.setProp(key, value);
+	}
+
+	@Override
+	public String getProperty(String key) {
+		return molecule.getProp(key);
+	}
+
+	@Override
+	public String[] getPropertyKeys() {
+		if (this.molecule != null) {
+			List<String> propertyKeys = new ArrayList<String>();
+			Str_Vect propList = this.molecule.getPropList();
+			for (int i = 0; i < propList.size(); i++) {
+				propertyKeys.add(propList.get(i));
+			}
+			String[] keys = new String[propertyKeys.size()];
+			return propertyKeys.toArray(keys);
+		} else {
+			return new String[0];
+		}
+	}
+
+	@Override
+	public String getPropertyType(String key) {
+		// Does not exist
+		return null;
+	}
+
+	@Override
+	public String getMolStructure() throws CmpdRegMolFormatException {
+		return this.molecule.MolToMolBlock();
+	}
+
+	@Override
+	public String getFormula() {
+		return RDKFuncs.calcMolFormula(this.molecule);
+
+	}
+
+	@Override
+	public Double getExactMass() {
+		return RDKFuncs.calcExactMW(this.molecule);
+	}
+
+	@Override
+	public Double getMass() {
+		return RDKFuncs.calcAMW(this.molecule);
+
+	}
+
+	@Override
+	public int getTotalCharge() {
+		return RDKFuncs.getFormalCharge(this.molecule);
+
+	}
+
+	@Override
+	public String getSmiles() {
+		return RDKFuncs.MolToSmiles(this.molecule);
+
+	}
+
+	@Override
+	public CmpdRegMolecule replaceStructure(String newStructure) throws CmpdRegMolFormatException {
+		CmpdRegMoleculeRDKitImpl newMol = new CmpdRegMoleculeRDKitImpl(newStructure);
+		Str_Vect propList = this.molecule.getPropList();
+		for (int i = 0; i < propList.size(); i++) {
+			newMol.setProperty(propList.get(i), this.getProperty(propList.get(i)));
+		}
+		return newMol;
+	}
+
+	@Override
+	public String getMrvStructure() {
+		// not implemented in RDkit
+		return null;
+	}
+
+	@Override
+	public byte[] toBinary(CmpdRegMolecule molecule, String imageFormat, String hSize, String wSize)
+			throws IOException {
+
+		// Read the preprocessor settings as json
+		ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Fully reliant on a service to do this right now.  No native RDkit functions implemented.